### PR TITLE
Add workflow for bal test native build

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Related: [#3316](https://github.com/ballerina-platform/ballerina-standard-library/issues/3316)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: ballerina-tests/build.gradle [hunk 1]
```diff
@@ -99,9 +114,9 @@ task ballerinaTest {
             workingDir project.projectDir
             environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
             if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                commandLine 'cmd', '/c', "${balJavaDebugParam} ${distributionBinPath}/bal.bat test ${testCoverageParam} --offline ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
+                commandLine 'cmd', '/c', "${balJavaDebugParam} ${distributionBinPath}/bal.bat test ${testParams} --offline ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
```
Comment (TharmiganK) [inline ballerina-tests/build.gradle:117]:
```suggestion
                commandLine 'cmd', '/c', "${balJavaDebugParam} ${distributionBinPath}/bal.bat test ${nativeFlag} ${testParams} --offline ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
```

File: ballerina-tests/build.gradle [hunk 2]
```diff
@@ -99,9 +114,9 @@ task ballerinaTest {
...
             } else {
-                commandLine 'sh', '-c', "${balJavaDebugParam} ${distributionBinPath}/bal test ${testCoverageParam} --offline ${groupParams} ${disableGroups} ${debugParams}"
+                commandLine 'sh', '-c', "${balJavaDebugParam} ${distributionBinPath}/bal test ${testParams} --offline ${groupParams} ${disableGroups} ${debugParams}"
```
Comment (TharmiganK) [inline ballerina-tests/build.gradle:119]:
```suggestion
                commandLine 'sh', '-c', "${balJavaDebugParam} ${distributionBinPath}/bal test ${nativeFlag} ${testParams} --offline ${groupParams} ${disableGroups} ${debugParams}"
```

**Comment by TharmiganK** [comment]
Can we also add `--native` flag to `examples` sub-project?

**Comment by ThisaruGuruge** [comment]
> Can we also add `--native` flag to `examples` sub-project?

Fixed with d9b796736620eea01152a147bb3596a2e157c583

File: examples/build.gradle [hunk 3]
```diff
@@ -46,9 +46,9 @@ task testExamples {
                 exec {
                     workingDir project.projectDir
                     if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                        commandLine 'cmd', '/c', "${ballerinaDist}/bin/bal.bat test --offline ${example} && exit %%ERRORLEVEL%%"
+                        commandLine 'cmd', '/c', "${ballerinaDist}/bin/bal.bat test --native --offline ${example} && exit %%ERRORLEVEL%%"
```
Comment (TharmiganK) [inline examples/build.gradle:49]: Suggested to have the flag only when `balNativeTest` property is set

```gradle
def nativeFlag = ""

task testExamples {
    if (project.hasProperty('balNativeTest')) {
         nativeFlag = '--native'
     }
    doLast {
        examples.each { example ->
            try {
                exec {
                    workingDir project.projectDir
                    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
                        commandLine 'cmd', '/c', "${ballerinaDist}/bin/bal.bat test ${nativeFlag} --offline ${example} && exit %%ERRORLEVEL%%"
                    } else {
                        commandLine 'sh', '-c', "${ballerinaDist}/bin/bal test ${nativeFlag} --offline ${example}"
                    }
                }
            } catch (Exception e) {
                println("Example '${example}' Build failed: " + e.message)
                throw e
            }
        }
    }
}
```

Comment (ThisaruGuruge) [reply examples/build.gradle]: My bad, sorry. Updated the PR.

**Review by TharmiganK — APPROVED** [review]

**Pull request merged**